lib/fluent/plugin/out_google_cloud.rb

```diff
@@ -958,50 +1030,69 @@ def decode_cloudfunctions_function_name(function_name)
     end
 
     # Requires that record has a 'kubernetes' field.
-    def handle_container_metadata(record, entry)
-      fields = %w(namespace_id namespace_name pod_id pod_name container_name)
-      fields.each do |field|
-        field_to_label(record['kubernetes'], field, entry.metadata.labels,
-                       "#{CONTAINER_SERVICE}/#{field}")
+    def extract_container_metadata(record)
+      resource_labels = {}
+      common_labels = {}
+      %w(namespace_id pod_id container_name).each do |field|
+        resource_labels.merge!(
+          field_to_label(record['kubernetes'], field, field))
+      end
+      %w(namespace_name pod_name).each do |field|
+        common_labels.merge!(
+          field_to_label(record['kubernetes'], field,
```

Comment:
If you change `field_to_label` to `fields_to_labels` as I suggested above, you could also merge these two loops into one call.